Lib/inspect.py

```diff
@@ -378,11 +378,16 @@ def isfunction(object):
     Function objects provide these attributes:
         __doc__         documentation string
         __name__        name with which this function was defined
+        __qualname__    fully qualified name of this function
+        __module__      name of the module the function was defined in or None
         __code__        code object containing compiled function bytecode
         __defaults__    tuple of any default values for arguments
         __globals__     global namespace in which this function was defined
         __annotations__ dict of parameter annotations
-        __kwdefaults__  dict of keyword only parameters with defaults"""
+        __kwdefaults__  dict of keyword only parameters with defaults
+        __dict__        namespace which is supporting arbitrary function attributes
+        __closure__     None or tuple of cells
```

Comment:
Personal suggestion:

```suggestion
        __closure__     a tuple of cells or None
```

I always find it weird to have "None" in the first place. Same when I see `None is [...]` or `NULL == ...`. Also, it seems that other attributes have "or None" at the end.